bundles/persistence/org.openhab.persistence.influxdb/src/main/java/org/openhab/persistence/influxdb/internal/InfluxDBPersistenceService.java

```diff
@@ -0,0 +1,582 @@
+  private InfluxDB influxDB;
+  private static final Logger logger = LoggerFactory.getLogger(InfluxDBPersistenceService.class);
+  private static final String TIME_COLUMN_NAME = "time";
+  private static final TimeUnit timeUnit = TimeUnit.MILLISECONDS;
+  private String dbName;
+  private String url;
+  private String user;
+  private String password;
+  private boolean isProperlyConfigured;
+  private boolean connected;
+
+  public void setItemRegistry(ItemRegistry itemRegistry) {
+    this.itemRegistry = itemRegistry;
+  }
+
+  public void unsetItemRegistry(ItemRegistry itemRegistry) {
+    this.itemRegistry = null;
+  }
+
+  public void activate(final BundleContext bundleContext, final Map<String, Object> config) {
+    logger.debug("influxdb persistence service activated");
+    disconnect();
+    password = (String) config.get("password");
+    if (StringUtils.isBlank(password)) {
+      isProperlyConfigured = false;
+      logger
+          .error("influxdb:password",
+              "The password is missing. To specify a password configure the password parameter in openhab.cfg.");
+      return;
+    }
+
+    url = (String) config.get("url");
+    if (StringUtils.isBlank(url)) {
+      url = DEFAULT_URL;
+      logger.debug("using default url {}", DEFAULT_URL);
+    }
+
+    user = (String) config.get("user");
+    if (StringUtils.isBlank(user)) {
+      user = DEFAULT_USER;
+      logger.debug("using default user {}", DEFAULT_USER);
+    }
+
+    dbName = (String) config.get("db");
+    if (StringUtils.isBlank(dbName)) {
+      dbName = DEFAULT_DB;
+      logger.debug("using default db name {}", DEFAULT_DB);
+    }
+
+    isProperlyConfigured = true;
+
+    connect();
+
+    // check connection; errors will only be logged, hoping the connection will work at a later
+    // time.
+    if (!checkConnection()) {
+      logger.error("database connection does not work for now, will retry to use the database.");
+    }
+  }
+
+  public void deactivate() {
+    logger.debug("influxdb persistence service deactivated");
+    disconnect();
+  }
+
+  private void connect() {
+    if (influxDB == null) {
+      // reuse an existing InfluxDB object because concerning the database it has no state
+      // connection
+      influxDB = InfluxDBFactory.connect(url, user, password);
+      influxDB.enableBatch(200, 100, timeUnit);
+    }
+    connected = true;
+  }
+
+  private boolean checkConnection() {
+    boolean dbStatus = false;
+    if (!connected) {
+      logger.error("checkConnection: database is not connected");
+      dbStatus = false;
+    } else {
+      try {
+        Pong pong = influxDB.ping();
+        String version = pong.getVersion();
+        // may be check for version >= 0.9
+        if (version != null && !version.contains("unknown")) {
+          dbStatus = true;
+          logger.debug("database status is OK");
+        } else {
+          logger.error("database connection failed, version is: \"{}\" response time was \"{}\"",
```

Comment:
is correct to log "connection failed"? At this point we can only say that the ping failed, right? A physical connection could be available though.